### PR TITLE
omit: perf improvements, better api, export from index

### PR DESCRIPTION
**Pull request by dzearing**
Changes omit to use `for in` loop and array/indexOf for matching exclusions. This seems fastest for small sets of exclusions (and is also the most ergonomic api.)

Updated jsperf test here: https://jsperf.com/omit-vs-rest-vs-reduce/10

image

I'm really confused why my initial tests were not reporting indexOf faster, but now i'm seeing that being faster.

Also realized that the omit I recently checked in was not being exported. Changing the API to take in an array of exclusions first.

### CODE REVIEW

File: packages/utilities/src/object.ts [hunk 1]
```diff
@@ -115,15 +115,15 @@ export function values<T>(obj: any): T[] {
...
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function omit<TObj extends Record<string, any>>(obj: TObj, exclusions: Record<string, any>): TObj {
+export function omit<TObj extends Record<string, any>>(obj: TObj, exclusions: (keyof TObj)[]): TObj {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const result: Record<string, any> = {};
 
-  for (const key of Object.keys(obj)) {
-    if (!exclusions[key]) {
+  for (const key in obj) {
```
Comment (xugao) [inline packages/utilities/src/object.ts:125]: why this change?

Comment (dzearing) [reply packages/utilities/src/object.ts:125]: see perf numbers.
